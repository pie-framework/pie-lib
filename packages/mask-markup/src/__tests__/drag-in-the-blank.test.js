import * as React from 'react';
import { shallow } from 'enzyme';
import { DragInTheBlank } from '../index';

const markup = `<div>
  <img src="https://image.shutterstock.com/image-vector/cow-jumped-over-moon-traditional-260nw-1152899330.jpg"></img>
   <h5>Hey Diddle Diddle <i>by ?</i></h5>
 <p>1: Hey, diddle, diddle,</p>
 <p>2: The cat and the fiddle,</p>
 <p>3: The cow {{0}} over the moon;</p>
 <p>4: The little dog {{1}},</p>
 <p>5: To see such sport,</p>
 <p>6: And the dish ran away with the {{2}}.</p>
</div>`;
const choice = (v, id) => ({ value: v, id });

describe('DragInTheBlank', () => {
  const defaultProps = {
    disabled: false,
    feedback: {},
    markup,
    choices: [
      choice('Jumped', '0'),
      choice('Laughed', '1'),
      choice('Spoon', '2'),
      choice('Fork', '3'),
      choice('Bumped', '4'),
      choice('Smiled', '5')
    ],

    value: {
      0: undefined
    }
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DragInTheBlank {...defaultProps} />);
  });

  describe('render', () => {
    it('renders correctly with default props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with disabled prop as true', () => {
      wrapper.setProps({ disabled: true });
      expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with feedback', () => {
      wrapper.setProps({
        feedback: {
          0: {
            value: 'Jumped',
            correct: 'Jumped'
          },
          1: {
            value: 'Laughed',
            correct: 'Laughed'
          },
          2: {
            value: 'Spoon',
            correct: 'Spoon'
          }
        }
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
