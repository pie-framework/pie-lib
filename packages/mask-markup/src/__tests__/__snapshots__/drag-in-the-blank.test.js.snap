// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragInTheBlank render renders correctly with default props 1`] = `
<div>
  <Choices
    disabled={false}
    value={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
  />
  <WithMask
    choices={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
    disabled={false}
    feedback={Object {}}
    markup="<div>
  <img src=\\"https://image.shutterstock.com/image-vector/cow-jumped-over-moon-traditional-260nw-1152899330.jpg\\"></img>
   <h5>Hey Diddle Diddle <i>by ?</i></h5>
 <p>1: Hey, diddle, diddle,</p>
 <p>2: The cat and the fiddle,</p>
 <p>3: The cow {{0}} over the moon;</p>
 <p>4: The little dog {{1}},</p>
 <p>5: To see such sport,</p>
 <p>6: And the dish ran away with the {{2}}.</p>
</div>"
    value={
      Object {
        "0": undefined,
      }
    }
  />
</div>
`;

exports[`DragInTheBlank render renders correctly with disabled prop as true 1`] = `
<div>
  <Choices
    disabled={true}
    value={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
  />
  <WithMask
    choices={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
    disabled={true}
    feedback={Object {}}
    markup="<div>
  <img src=\\"https://image.shutterstock.com/image-vector/cow-jumped-over-moon-traditional-260nw-1152899330.jpg\\"></img>
   <h5>Hey Diddle Diddle <i>by ?</i></h5>
 <p>1: Hey, diddle, diddle,</p>
 <p>2: The cat and the fiddle,</p>
 <p>3: The cow {{0}} over the moon;</p>
 <p>4: The little dog {{1}},</p>
 <p>5: To see such sport,</p>
 <p>6: And the dish ran away with the {{2}}.</p>
</div>"
    value={
      Object {
        "0": undefined,
      }
    }
  />
</div>
`;

exports[`DragInTheBlank render renders correctly with feedback 1`] = `
<div>
  <Choices
    disabled={false}
    value={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
  />
  <WithMask
    choices={
      Array [
        Object {
          "id": "0",
          "value": "Jumped",
        },
        Object {
          "id": "1",
          "value": "Laughed",
        },
        Object {
          "id": "2",
          "value": "Spoon",
        },
        Object {
          "id": "3",
          "value": "Fork",
        },
        Object {
          "id": "4",
          "value": "Bumped",
        },
        Object {
          "id": "5",
          "value": "Smiled",
        },
      ]
    }
    disabled={false}
    feedback={
      Object {
        "0": Object {
          "correct": "Jumped",
          "value": "Jumped",
        },
        "1": Object {
          "correct": "Laughed",
          "value": "Laughed",
        },
        "2": Object {
          "correct": "Spoon",
          "value": "Spoon",
        },
      }
    }
    markup="<div>
  <img src=\\"https://image.shutterstock.com/image-vector/cow-jumped-over-moon-traditional-260nw-1152899330.jpg\\"></img>
   <h5>Hey Diddle Diddle <i>by ?</i></h5>
 <p>1: Hey, diddle, diddle,</p>
 <p>2: The cat and the fiddle,</p>
 <p>3: The cow {{0}} over the moon;</p>
 <p>4: The little dog {{1}},</p>
 <p>5: To see such sport,</p>
 <p>6: And the dish ran away with the {{2}}.</p>
</div>"
    value={
      Object {
        "0": undefined,
      }
    }
  />
</div>
`;
